src/scripts/convertBreachResolutions.js

```
@@ -0,0 +1,97 @@
+
+// find all subscribers who resolved any breaches in the past, convert those
+// records into the new v2 format
+do {
+  console.log(`Converting breadches_resolved to breach_resolution - start: ${offset} limit: ${LIMIT}`)
+  subscribersArr = await knex
+    .select('id', 'primary_email', 'breaches_resolved', 'breach_resolution')
+    .from('subscribers')
+    .whereNotNull('breaches_resolved')
+    .limit(LIMIT)
+    .offset(offset)
+
+  console.log(`Loaded # of subscribers: ${subscribersArr.length}`)
+
+  for (const subscriber of subscribersArr) {
+    const { breaches_resolved: v1, breach_resolution: v2 } = subscriber
+    console.debug({ v1 })
+    console.debug({ v2 })
+
+    let isV2Changed = false // use a boolean to track if v2 has been changed, only upsert if so
+
+    // fetch subscriber all breaches / email
+    const subscriberBreachesEmail = await getAllEmailsAndBreaches(subscriber, allBreaches)
+    console.debug(JSON.stringify(subscriberBreachesEmail.verifiedEmails))
+
+    for (const [email, resolvedRecencyIndices] of Object.entries(v1)) {
+      console.debug({ email })
+      console.debug({ resolvedRecencyIndices })
+      for (const recencyIndex of resolvedRecencyIndices) {
+        console.debug({ recencyIndex })
+        // find subscriber's relevant recency index breach information
+        const ve = subscriberBreachesEmail.verifiedEmails?.filter(ve => ve.email === email)[0] || {}
+        const subBreach = ve.breaches?.filter(b => Number(b.recencyIndex) === Number(recencyIndex))[0] || null
+        console.debug({ subBreach })
+
+        if (!subBreach || !subBreach.DataClasses) {
+          console.warn(`SKIP: Cannot find subscribers breach and data types - recency: ${recencyIndex} email: ${email}`)
+          continue
+        }
+
+        // if email does not exist in v2, we need to add it to the object
+        // format: {email: { recencyIndex: { isResolved: true, resolutionsChecked: [DataTypes]}}}
+        if (!v2[email]) {
```

Because `v2` is null for me locally, the script fails at this point.  We could add optional chaining `?.` but I'm not sure that fixes the root issue?  Should `v2` first be instantiated as an object maybe?

---

See comment above. We don't want to init, we should let it fail. I'll add another check here for `!v2 ||`

---

Oh nvm I see what you are saying